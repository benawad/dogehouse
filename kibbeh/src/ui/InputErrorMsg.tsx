import React from "react";

interface InputErrorMsgProps {}

// @todo verify with designer what color this should be
export const InputErrorMsg: React.FC<InputErrorMsgProps> = ({ children }) => {
  return (
    <div className={`flex text-secondary`} data-testid="input-error-msg">
      {children}
    </div>
  );
};
