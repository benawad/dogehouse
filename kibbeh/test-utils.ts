import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
