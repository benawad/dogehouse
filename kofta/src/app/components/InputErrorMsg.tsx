import React from "react";

interface InputErrorMsgProps {}

export const InputErrorMsg: React.FC<InputErrorMsgProps> = ({ children }) => {
  return <div className={`text-red-600`}>{children}</div>;
};
