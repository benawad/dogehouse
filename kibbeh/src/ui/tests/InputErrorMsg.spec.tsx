import React from "react";

import { render } from "../../../test-utils";
import { InputErrorMsg } from "../InputErrorMsg";

describe("InputErrorMsg", () => {
  it("should render correctly", () => {
    const { getByTestId } = render(<InputErrorMsg />);
    const component = getByTestId("input-error-msg");

    expect(component).toBeVisible();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<InputErrorMsg />);
    const component = getByTestId("input-error-msg");
    expect(component).toMatchSnapshot();
  });
});
