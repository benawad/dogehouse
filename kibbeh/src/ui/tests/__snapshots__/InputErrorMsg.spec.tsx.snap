// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputErrorMsg should match snapshot 1`] = `
<div
  class="flex text-secondary"
  data-testid="input-error-msg"
/>
`;
